src/transport/tcp/config.rs

```diff
@@ -80,6 +80,12 @@ pub struct Config {
     /// How long should litep2p wait for a substream to be opened before considering
     /// the substream rejected.
     pub substream_open_timeout: std::time::Duration,
+
+    /// DNS resolver config.
+    pub resolver_config: hickory_resolver::config::ResolverConfig,
+
+    /// DNS resolver options.
+    pub resolver_opts: hickory_resolver::config::ResolverOpts,
```

Comment:
Ideally this should be universal to all the transports, including WS, QUIC, etc.

Reply:
Sorry, missed this. Yes, `TransportBuilder:new` accepting a new argument is fine.